lib/promscrape/discovery/kubernetes/endpoints_test.go

```diff
@@ -115,8 +115,9 @@ func TestParseEndpointsListSuccess(t *testing.T) {
 
 func TestGetEndpointsLabels(t *testing.T) {
 	type testArgs struct {
-		containerPorts map[string][]ContainerPort
-		endpointPorts  []EndpointPort
+		containerPorts     map[string][]ContainerPort
+		initContainerPorts map[string][]ContainerPort
```

Comment:
I suggest to introduce additional in-place type initContainerPorts:

```
type initContainerWithPorts{
  name string
  image string
  restartPolicy string
  ports []ContainerPort
}
```
It should improve test quality.